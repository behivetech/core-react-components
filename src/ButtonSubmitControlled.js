// Vendor Libs
import React, {Component} from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import {omit} from 'lodash';

// Components
import Button from './Button';
import withFormControl from './withFormControl';

class ButtonSubmitControlled extends Component {
    render() {
        const {className, disabled, formState} = this.props;
        const props = {
            ...omit(this.props, ['formState']),
            className: classnames('button-submit-controlled', className),
            disabled: disabled || formState.formDisabled,
            type: 'submit',
        };

        return <Button {...props} />;
    }
}

ButtonSubmitControlled.propTypes = {
    /** Ability to add additional className to the component for addistional styling from parent */
    className: PropTypes.string,
    /** Disables button if true. */
    disabled: PropTypes.bool,
    formState: PropTypes.shape({
        /** Comes from the state of the FormControlled component. Disables button if true. */
        formDisabled: PropTypes.bool,
    }),
};

export default withFormControl(ButtonSubmitControlled);