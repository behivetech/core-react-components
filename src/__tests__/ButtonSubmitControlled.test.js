import ButtonSubmitControlled from '../ButtonSubmitControlled';

jest.mock('../withFormControl');

describe('ButtonSubmitControlled', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should render with the correct props', () => {
        const wrapper = shallow(<ButtonSubmitControlled className="mock-class" />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.dive()).toMatchSnapshot();
        wrapper.setProps({type: 'submit', children: undefined});
        expect(wrapper.dive()).toMatchSnapshot();
    });

    it('should disable properly', () => {
        const wrapper = shallow(<ButtonSubmitControlled className="mock-class" />).dive();

        wrapper.setProps({diabled: true});
        expect(wrapper).toMatchSnapshot();
        wrapper.setProps({diabled: false, formstate: {formDisabled: true}});
        expect(wrapper).toMatchSnapshot();
    });
});