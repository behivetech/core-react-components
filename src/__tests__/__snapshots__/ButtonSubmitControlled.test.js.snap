// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonSubmitControlled should disable properly 1`] = `
<Button
  className="button-submit-controlled mock-class"
  diabled={true}
  disabled={false}
  onClick={[Function]}
  type="submit"
/>
`;

exports[`ButtonSubmitControlled should disable properly 2`] = `
<Button
  className="button-submit-controlled mock-class"
  diabled={false}
  disabled={false}
  formstate={
    Object {
      "formDisabled": true,
    }
  }
  onClick={[Function]}
  type="submit"
/>
`;

exports[`ButtonSubmitControlled should render with the correct props 1`] = `
<ButtonSubmitControlled
  className="mock-class"
  formState={
    Object {
      "formDisabled": false,
      "getFieldError": [MockFunction],
      "getFieldValue": [MockFunction],
      "initialSubmit": false,
      "setFieldValidators": [MockFunction],
      "setFieldValue": [MockFunction],
      "setFieldValueDebounced": [MockFunction],
      "shouldDisable": [MockFunction],
    }
  }
/>
`;

exports[`ButtonSubmitControlled should render with the correct props 2`] = `
<Button
  className="button-submit-controlled mock-class"
  disabled={false}
  onClick={[Function]}
  type="submit"
/>
`;

exports[`ButtonSubmitControlled should render with the correct props 3`] = `
<Button
  className="button-submit-controlled mock-class"
  disabled={false}
  onClick={[Function]}
  type="submit"
/>
`;
